scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -701,6 +704,20 @@ impl TomlManifest {
             })
             .transpose()
     }
+
+    fn collect_fmt_config(&self, tool: &Option<TomlToolsDefinition>) -> Result<ManifestFmtConfig> {
```

Comment:
I gave this PR a thought and I think this code does it wrong 😞 I didn't investigate the context when outlining implementation steps; my bad.
1. We should stick with original naming, so this field should be named `sort-module-level-items`, for consistency.
2. I don't think we should trigger errors when reading the manifest, if user has errors in `[tool.*]` sections. These should pop only in affected tools, IMHO. This code actually does not validate the values at all, it just ignores them.
3. There is a way how to deduplicate stuff nicely!

So first of all, do note that Cairo's `FormatterConfig` has `Default` implementation. I think we should submit a PR which adds the following for this struct:

```rust
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "kebab-case")]
pub struct FormatterConfig {
    tab_size: usize,
    max_line_length: usize,
    sort_module_level_items: bool,
}
```

This way we will keep upstream 1) default values 2) canonical serde of this struct.

Now, about parsing. We have very nifty methods in our `Package` struct: https://github.com/software-mansion/scarb/blob/d98fca8190ff330109b414ed70634b569732a6b5/scarb/src/core/package/mod.rs#L92-L113

We also have this very useful helper function: https://github.com/software-mansion/scarb/blob/c412b06b0e1069ea1ca503c945d5e8dfd12bb398/scarb/src/internal/serdex.rs#L8-L25

What we can do with this the following. All of this will happen within `fmt` op **only**!

```rust
let mut config = FormatterConfig::default();
if let Some(overrides) = package.tool_metadata("fmt") {
    config = toml_merge(&config, overrides)?;
}
```

_voila!_